shared/actions/fs/shared.js

```diff
@@ -14,16 +14,23 @@ const _getRouteChangeActionForPermissionError = (path: Types.Path) =>
   RouteTreeGen.createNavigateTo({path: [...fsRootRoute, {props: {path}, selected: 'oopsNoAccess'}]})
 
 const makeErrorHandler = (action: FsGen.Actions, retriable: boolean) => (error: any): TypedActions => {
-  if (
-    (typeof error.desc === 'string' && error.desc.includes('does not have read access to')) ||
-    error.desc.includes('You are not a member of this team')
-  ) {
-    // TODO: add a real error code for this
-    return _getRouteChangeActionForPermissionError(
-      // If you get a flow error here after aadding an action that has a 'path'
-      // field, try rename that field if the field is not of type FsTypes.Path.
-      (action.payload && action.payload.path) || Constants.defaultPath
-    )
+  // TODO: add and use proper error code for all these
+  if (typeof error.desc === 'string') {
+    if (
+      error.desc.includes('does not have read access to') ||
+      error.desc.includes('You are not a member of this team') ||
+      // team doesn't exist
+      error.desc.includes('Root team does not exist') ||
+      // public tlf doesn't exist
+      error.desc.includes("Can't create TLF ID for non-team-backed handle")
+    ) {
+      // TODO: add a real error code for this
+      return _getRouteChangeActionForPermissionError(
+        // If you get a flow error here after aadding an action that has a 'path'
+        // field, try rename that field if the field is not of type FsTypes.Path.
+        (action.payload && action.payload.path) || Constants.defaultPath
+      )
+    }
```

Comment:
Not sure if this was supposed to display the Oops popup, but it's still hanging on my machine when I go to `/keybase/private/zila.test/abc/nonexistent.txt`

Reply:
With this branch? That's expected; https://github.com/keybase/client/pull/15426 is the one that fixes the non-existent thing.